crates/usvg/src/writer.rs

```diff
@@ -1228,3 +1369,274 @@ fn write_num(num: f32, buf: &mut Vec<u8>, precision: u8) {
+fn write_span(
+    is_clip_path: bool,
+    writer_context: &mut WriterContext,
+    xml: &mut XmlWriter,
+    text: &Text,
+    char_offset: &mut usize,
+    chunk: &TextChunk,
+    span: &TextSpan,
+) {
+    xml.start_svg_element(EId::Tspan);
+
+    write_font_family(xml, span);
+    write_font_style(xml, span);
+    write_font_weight(xml, span);
+    write_font_size(xml, span);
+
+    write_visibility(xml, span);
+    write_letter_spacing(xml, span);
+    write_word_spacing(xml, span);
+    write_text_length(xml, span);
+    write_length_adjust(xml, span);
+
+    write_smallcaps(xml, span);
+
+    write_paint_order(xml, span);
+
+    write_kerning(xml, span);
+
+    write_alignment_baseline(xml, span);
+    write_dominant_baseline(xml, span);
+
+    write_fill(&span.fill, is_clip_path, writer_context.opt, xml);
+    write_stroke(&span.stroke, writer_context.opt, xml);
+
+    let cur_text = &chunk.text[span.start..span.end];
+
+    let actual_start = *char_offset + span.start;
+    let num_chars = cur_text.chars().count();
+    let actual_end = *char_offset + span.start + num_chars;
+
+    write_coordinates(xml, text, actual_start, actual_end);
+    write_rotations(xml, text, actual_start, actual_end);
+    xml.write_text(&cur_text.replace('&', "&amp;"));
+
+    xml.end_element();
+}
+
+fn write_kerning(xml: &mut XmlWriter, span: &TextSpan) {
+    if !span.apply_kerning {
+        xml.write_attribute_raw("style", |buf| {
+            buf.extend_from_slice("font-kerning:none".as_bytes())
+        });
+    }
+}
+
+fn write_paint_order(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.paint_order == PaintOrder::StrokeAndFill {
+        xml.write_svg_attribute(AId::PaintOrder, "stroke fill");
+    }
+}
+
+fn write_smallcaps(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.small_caps {
+        xml.write_svg_attribute(AId::FontVariant, "small-caps");
+    }
+}
+
+fn write_length_adjust(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.length_adjust == LengthAdjust::SpacingAndGlyphs {
+        xml.write_svg_attribute(AId::LengthAdjust, "spacingAndGlyphs");
+    }
+}
+
+fn write_text_length(xml: &mut XmlWriter, span: &TextSpan) {
+    if let Some(text_length) = span.text_length {
+        xml.write_svg_attribute(AId::TextLength, &text_length);
+    }
+}
+
+fn write_word_spacing(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.word_spacing != 0.0 {
+        xml.write_svg_attribute(AId::WordSpacing, &span.word_spacing);
+    }
+}
+
+fn write_letter_spacing(xml: &mut XmlWriter, span: &TextSpan) {
+    if span.letter_spacing != 0.0 {
+        xml.write_svg_attribute(AId::LetterSpacing, &span.letter_spacing);
+    }
+}
+
+fn write_baseline_shift(xml: &mut XmlWriter, baseline_shift: &BaselineShift) {
+    match baseline_shift {
+        BaselineShift::Number(num) => xml.write_svg_attribute(AId::BaselineShift, num),
+        BaselineShift::Baseline => xml.write_svg_attribute(AId::BaselineShift, "baseline"),
```

Comment:
There is no `baseline` in SVG/CSS. It should be ignored completely in this case. `BaselineShift::Baseline` is basically `None`. But it's simpler to have `BaselineShift::Baseline` compared to `Option<BaselineShift>`.